### ISSUE TITLE
Add content warning on a boost

### ISSUE THREAD

**Issue by rbelu** [mastodon/mastodon#13291]
Give the opportunity to add a CW when you boost a toot which doesn't have already a CW

### Pitch
As a user, i'd like to have the opportunity to add a CW to a toot when I boost it.

### Motivation
CW rules are different in the instances of the fediverse. Sometimes, i'd like to boost a toot which have no CW but contains mention of violence, rape, ... 

On the instance where I have my accounts violence, rape .. should be hide behind a CW so I don't share interesting content because of that

**Comment by SuperSandro2000** [comment]
This would alter the toot you are boosting and I don't like the idea of that.

**Comment by bubblineyuri** [comment]
This really looks just another interpretation of the "quote tweet" feature of Twitter which is the main reason why people use Mastodon.

**Comment by rbelu** [comment]
I understand the point. But, you can actually quote a toot by sharing its
link which seems to be worst than the feature I propose because the author
of the initial toot is not notified.

I won't talk for other people, but on my side i did not leave twitter
because of the quote feature but because of their uge lack of moderation
and CW are part of the response to it.

**Comment by Cassolotl** [comment]
Kind of a duplicate of *Feature request: Boost with a CW #2719*, where Gargron said "(Essentially) duplicate of #309".

**Comment by davidak** [comment]
And #309 was closed because it could be used for harassment.

https://github.com/tootsuite/mastodon/issues/309#issuecomment-469260847

But i think there could be technical solutions found to prevent this.

For CWs, could Mastodon include default categories where a user can categorize their toots under? Could there be a consensus of topics that should have a CW? That would also help users to decide if a CW would be good. They could be recommended while composing a toot based on keywords. And also instance admins could add instance specific categories. Sure instance admins that promote harassment can include hateful categories like "Untermensch", but we block those instances, right?

@Gargron i need this feature right now because my instance urges everyone to CW all posts related to COVID-19. I can do that for my own posts, but not boosts! So i can't share important or funny posts unless i copy the content and create my own toot with CW which breaks the connection to the original (which might include useful comments).

https://chaos.social/@ordnung/103809178144073770

**Comment by rbelu** [comment]
Same on my instance, a lot of toot are linked to add CW on posts related to Covid-19 and Cop violence which are trending topics in France right now.

**Comment by bubblineyuri** [comment]
Personally I would only give this change a try if there was an actual indication that the boosted toot has been altered e.g. a small pencil icon and if you hover over it it will give you the following information

"This boost has been altered to include a content warning. The original post may reflect different views."

**Comment by limdingwen** [comment]
A few ideas relating to this:

1. Reply to the original post: “Boosting with CW”, with a CW, leaving out the ping. Then boost the reply. This is indeed rather tedious though.

2. If you’re a server admin and the issue is more with images, one can filter media coming from the server. 

I agree with the OP. CW rules are very different across the Fediverse and makes it stressful to boost at times (“aaa I really want to share this, but it’s not CW’d to my instance’s rules… what do I do?”). At the same time, it is true that this may be abused (eg boosting with an CW intending to insult the original post).

**Comment by daedhir** [comment]
I was just about to file a request for this. I just perused a couple of rule sets for a couple of instances today and was surprised about how much stuff on some instances requires a CW. This effectively disallows anybody on those instances to boost a lot of content.

It seems like more win than loss to allow people to attach a CW to boosts.

**Comment by idoric** [comment]
With the recent arrival of quite a few friends, I found myself thinking about the question. I understood the danger of seeing such a feature used as a "Quote Retweet".

It was suggested in a closed ticket to limit CWs to a predefined closed list of tags¹. I have the feeling that we would lose a large part of the interest of the CW, but above all it would be quite possible to use a tag in an ironic way, or to twist its meaning.

The more I think about it, the more I say to myself that if we want to make sure that the will of the author of the toot is respected, the most obvious and the simplest thing is to ask him! This would take the form of a notification specifying which CW would be added to the toot. If he refuses, nothing more happens, if he accepts, the message will be effectively boosted with the addition of the CW.

Even in the context of my suggestion, I agree with @bubblineyuri that it should be made very clear that the CW is a later addition.

¹ https://github.com/mastodon/mastodon/issues/2719#issuecomment-299521499

**Comment by neatchee** [comment]
+1 for this, requiring a technical solution.

This seems like it should be as simple as attaching metadata to a boost.
I would need to know exactly how boosting functions under the hood, but if I'm able to attach my username to a boost then there seems to be no reason that metadata couldn't include a "BoostCW" boolean flag. Even if that doesn't include a detail of why it's CW'd, that's better than not being able to CW boosts at all.

I'm not sure that I agree with the sentiment that this is "modifying the author's content" any more than copy-pasting the URL with a comment is "modifying the author's content". Commentary is not modification.

If the problem is replicating the QRT feature from birdsite, then the issue is very specifically the addition of a text comment on a boost, and not the marking of that boost with a CW flag.

**Comment by Crashdoom** [comment]
+1 for this.

There's a lot more content available now through the numerous relays that sprout up, and there's no clear consensus on how to address CWs. Some instances are more strict, some are less strict.

Currently admins and users lack the ability to tag content that would potentially violate local rules, that would otherwise not violate the originating instance's rules, and that puts admins in a poor situation:

- **Delete the offending content**: That immediately takes care of the issue, except now users on your instance who would like to see that content behind a CW can no longer interact with it. Users being able to favorite or boost from the originating instance is not a workaround.
- **Leave the content**: However, this introduces a double-standard on the admins where they are forced to keep offending content without a CW, while enforcing the rule for CWs to users on their instance. This generates strife between the admin team and their users.

I would agree with others that if the QRT system from birdsite is truly the issue, then there should be no problems with the addition of a CW with an appropriate icon on the boosted toot that identifies it as having been marked by a third-party and not the original poster. (See @bubblineyuri's comment).

But, this isn't something we can ignore as it's only going to cause more and more problems and lead to fracturing the fediverse because of these CW differences.

**Comment by Talik13** [comment]
This is just a thought - but I think a root cause of "ironic" content warnings is for the attention. Boosting a post is also kind of for the attention, but hear me out: What if, when adding a content warning to a boosted post, it simply directs anyone who wants to view it directly to the original author's post.

I know that some people wouldn't be okay with missing out on the "likes" and "boosts" they'd get from their boost. Personally, I would be okay with forfeiting the knowledge of "how many people liked what I liked" in order to respect other people's content triggers. 

PLUS: it has the added benefit of giving the original poster more genuine attention.

Negatives:
It can still technically be used for trolling, but the trolls would never actually receive a favorite or boost - thus making it less entertaining an option. And it also makes it MUCH easier to moderate on the instance side of things.

🤷

**Comment by limdingwen** [comment]
@Talik13, I like that idea. It's like how the "filter media"/"Not Available" function works.

**Comment by jfinkhaeuser** [comment]
+1

@Talik13 You can still add a UI to the original toot that reads "boosted by X,Y, Z and N others." or some such. It's not quite the same as favourites, but it gives some credit. Could even expand this to a full list like with polls.

FWIW, quote replies aren't as bad as everyone makes them. We've been using them on the Intarwebs for ages before the fediverse, and only with Twitter did they become something truly toxic. I would argue that the main reasons they're so bad on the birdsites are:

- They feed the rage algorithm, even if the comment is reasonable. There is no corresponding algorithm here.
- There is no UI on hellsite that collapses CW'd posts. Users *must* engage there, to a degree. Here? Not so much.

TL;DR IMHO the feature should be viewed within its own context, not a foreign one. That said, adding a CW is the kind of surgical "quote boost lite" that seems more likely to do good than harm.

**Comment by mashedkeyboard** [comment]
I very much like this idea, but wonder how it would be implemented ActivityPub-wise? The Announce Activity in ActivityPub wouldn't really permit this as I understand it, so this might require making a whole new Note.

**Comment by realpixelcode** [comment]
> Why do you want to add a content warning to this boost? Please select one or more reasons.
>
> - [ ] NSFW
> - [ ] visual gore
> - [ ] political content
> - [ ] disturbing text
> - [ ] spoilers (movie, book etc.)
> - [ ] other (display a generic CW)

**Comment by jfinkhaeuser** [comment]
Honestly, multiple choice seems like the wrong approach. For example, a common CW is food or drug related, and that's not in the list. You'll have a hard time capturing all the common use cases and remaining usable.

**Comment by realpixelcode** [comment]
Remember that this is only about boosts, not about CWs in general. I think CWs for boosts would still be beneficial even at the expense of not covering all use cases.

I mean, currently, we couldn't add a CW to a boosted food photo anyway, simply because boost CWs don't exist at all yet.

Not implementing boost CWs just because we couldn't possibly cover every use case imaginable would not be wise in my opinion.

Why should be deny boost CWs for *all use cases* just because there are a few that would not be covered?

**Comment by jfinkhaeuser** [comment]
That's the opposite of what I argue for. I think freeform CWs on boosts are sufficient (and necessary).

**Comment by Oblomov** [comment]
Content Warnings on boosts are possible according to the ActivityPub protocol specification. Boosts correspond to Announce activity, and any Activity can have a `summary` (which is what Mastodon uses as CW), so the Announce can have a CW which is _distinct_ from that of the announced (boosted) post. This also means that boosts with CW can (and should!) be implemented _without_ the controversial QT feature.

**Comment by trwnh** [comment]
Related to #13291

**Comment by dj-sf** [comment]
@Oblomov this is great insight.  I strongly support this approach.  Adding a content warning should be done in a way that does not alter the core content of the original post, and it sounds like we have the tools to make that possible.  Ironic content warnings are a legitimate concern but keep in mind that users have other more effective ways of being jerks if they choose to do so (such as posting a URL to the original post along with commentary).

If content warnings are controlled via metadata it could also be possible to allow a user boosting a post to remove a content warning that was not on the original post if they choose to do so (but not remove one that was on the original post).

Some background info: I'm interested in this feature because I want to make my mastodon account mostly about music content but occasionally include other kinds of content (mostly political in nature) that is SFW but could possibly trigger some viewers... especially if they came to my profile for music and are caught off guard.  I want the freedom to include all these kinds of content in a single profile without triggering users who are only in the mood for music.

**Comment by dj-sf** [comment]
also keep in mind that we can predict how users react to a feature change up to a certain point, but there will almost always be unforeseen consequences for any design decision.  

With this in mind, I believe we should start by implementing freeform content warnings for all boosts without exception, and monitor to see what happens.  I do not think we would see widespread or systemically problematic abuse of this feature... but if we did, we can always roll it back or alter it as needed..._especially_ if we keep the first iteration of the feature simple so that we don't waste tons of time and effort on something that does not end up working out.

**Comment by LivInTheLookingGlass** [comment]
Would a PR be welcome on this? I'm willing to do quite a bit of work if it gets this feature adopted

**Comment by neatchee** [comment]
> Would a PR be welcome on this? I'm willing to do quite a bit of work if it gets this feature adopted

PRs are always welcome. 

If you want the prototype tested, you might consider submitting to a fork like https://github.com/glitch-soc/mastodon first. Lots of early adopters of feature work there and folks like ClearlyClaire give great feedback that frequently results in features making their way upstream

**Comment by diondiondion** [comment]
This can be achieved with quote posts which were introduced in [Mastodon 4.5](https://github.com/mastodon/mastodon/releases/tag/v4.5.0), so we're closing this issue. If you have any concerns or feedback about the new quote posts, please open a new issue.

You can learn more about Quote posts in our [blog post](https://blog.joinmastodon.org/2025/09/introducing-quote-posts/), our [FAQ](https://docs.joinmastodon.org/user/quote-posts/#faq), and our [developer implementation guide](https://docs.joinmastodon.org/client/quotes/).

**Comment by trwnh** [comment]
If a post is not quotable, then this *cannot* "be achieved with quote posts".

**Comment by diondiondion** [comment]
> If a post is not quotable, then this _cannot_ "be achieved with quote posts".

Unfortunately adding arbitrary text content to a boost would provide an attack vector for circumventing missing quote approval. We've discussed this internally and decided this won't be implemented for this reason.

**Comment by ghost** [comment]
A lot of features people ask for are essentially GUI changes that provide a better user experience for what they'd now do manually.

We've had informal quote posts for a while now, by copy-pasting the link to the original post and commenting that.

We've had informal CW on boost for a while now, by copy-pasting the link to the original post and adding a CW.

There is no new attack vector here; that statement is an illusion.

It's also IMHO a dangerous illusion. By convincing yourself that this is fact rather than illusion, you also prevent the author of the original post from having *any* kind of influence on what is going on. They are not notified, they have no controls to mitigate things, etc.

You won't be able to prevent informal quote posts/CW on boost, unless your original post lacks a resolvable URL. If you want to close an attack vector, that's the only way to close it.

So what feature requests such as this and quote posts *really* are asking is "make this thing that we're doing anyway a little less shit in the user experience, please", with an additional "and think about how it may be abused instead of just ignoring that problem".

Your response essentially says "we're going to ignore the problem".

I mean, your decision, but let's not pretend it's anything else, please.
